File: Java/src/main/java/com/gildedrose/categories/BackstagePass.java
package com.gildedrose.categories;

import com.gildedrose.Item;

public class BackstagePass extends DefaultCategory {

    @Override
    public String getName() {
        return "Backstage passes to a TAFKAL80ETC concert";
    }

    @Override
    public void UpdateQuality(Item item) {
        if (item.sellIn < 0) {
            item.quality = 0;
        } else {
            if (item.sellIn < 5) {
                item.quality += 3;
            } else if (item.sellIn < 10) {
                item.quality += 2;
            } else {
                item.quality++;
            }
        }

        if (item.quality > 50) {
            item.quality = 50;
        }
    }
}

File: csharpcore/Categories/BackstagePass.cs
namespace csharpcore.Categories
{
    public class BackstagePass : DefaultCategory
    {
        public override string Name => "Backstage passes to a TAFKAL80ETC concert";

        public override void UpdateQuality(Item item)
        {
            if (item.SellIn < 0)
            {
                item.Quality = 0;
            }
            else
            {
                if (item.SellIn < 5)
                {
                    item.Quality += 3;
                }
                else if (item.SellIn < 10)
                {
                    item.Quality += 2;
                }
                else
                {
                    item.Quality++;
                }
            }

            if (item.Quality > 50)
            {
                item.Quality = 50;
            }
        }
    }
}